pdpipe/core.py

```diff
@@ -652,6 +655,17 @@ def __init__(self, stages, transformer_getter=None, **kwargs):
     def __getitem__(self, index):
         if isinstance(index, slice):
             return PdPipeline(self._stages[index])
+
+        if isinstance(index, list):
+            stages = [stage for stage in self._stages if stage._label in index]
+            return PdPipeline(stages)
+
+        if isinstance(index, str):
+            stages = [stage for stage in self._stages if stage._label == index]
+            if len(stages) == 0:
+                raise Exception("'{}' is not exist.".format(index))
```

Comment:
if python3.5 support was dropped, this could be in f-strings, which is quite more readable @shaypal5

Reply:
if two stages have the same label you will return just the first one in the pipeline. Maybe that's the desired behaviour, but highlighting just in case. Not sure whether we would like something when creating the pipeline to check for integrity (i.e. ensure that two stages cannot have same label), or we just trust the user

Reply:
> if python3.5 support was dropped, this could be in f-strings, which is quite more readable @shaypal5

I did not see the use of 'f-string' in the project, so I also did not use it.

> if two stages have the same label you will return just the first one in the pipeline. Maybe that's the desired behaviour, but highlighting just in case. Not sure whether we would like something when creating the pipeline to check for integrity (i.e. ensure that two stages cannot have same label), or we just trust the user

When passing in one item a `Stage` is returned, so I cant return two stages.
`pipeline[['someLabel']]` will return a pipeline with all stages that `label` == 'someLabel'.

Reply:
You're both right: I didn't f-strings so far, be we definitely should. I recently moved to using f-strings almost exclusively in my code. I'm also going into type hints, but I don't think it's a very urgent change.

Reply:
Regarding the second issue: See discussion below. A reasonable handling of this case should be decided upon and documented.

Reply:
re f-strings. We need python3.6 or be very determined and use this https://stackoverflow.com/questions/55182209/f-string-invalid-syntax-in-python-3-5 , which I wouldn't recommend @shaypal5

Reply:
Since Python 3.5 end-of-life was at September 2020, so Iet's do it!

Reply:
I think it's better to make a separate PR of dropping `python 3.5` and adding `f-strings`.